lang/cs/Org.Apache.REEF.Wake/Remote/Impl/TransportServer.cs

```diff
@@ -89,10 +80,45 @@ public IPEndPoint LocalEndpoint
         /// </summary>
         public void Run()
         {
-            _listener.Start();
+            if (LocalEndpoint.Port == 0)
+            {
+                FindAPortAndStartListener();
+            }
+            else
+            {
+                _listener.Start();
```

Comment:
How do we ensure that `_listener` isn't `null`? It is a mutable field, as per the discussion above.